src/app/user.service.ts

```diff
@@ -64,11 +64,20 @@ export class UsersService {
 		}
 	}
 
-	public deleteUser(userId: number) {
+	public deleteUser(userId: number): void {
 		const user: User | undefined = this.findUserById(userId);
-		if(user){
+		if (user) {
 			this.users.splice(this.users.indexOf(user), 1);
 		}
 	}
 
+	public isEmailAvailable(email: string): boolean {
+		for (let user of this.users) {
+			if (user.email == email) {
+				return false;
+			}
+		}
+		return true;
+	}
+
```

Comment:
you can try this:
isEmailTaken(email): boolean{

return this.users.some((user: User) => user.email === email);
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some